Dockerfile

```diff
@@ -1,29 +1,33 @@
 # syntax=docker/dockerfile:1
-
 FROM docker.io/nvidia/cuda:12.8.1-cudnn-devel-ubuntu22.04 AS builder
 
+ARG CUDA_COMPUTE_CAP=70
+ARG RAYON_NUM_THREADS=4
+ARG RUST_NUM_THREADS=4
+ARG RUSTFLAGS="-Z threads=${RUST_NUM_THREADS}"
```

Comment:
Why did you move all these? They were fine where they were before, co-located with the `RUN` for building the project where they're relevant. These don't need to be added to each earlier `RUN` step.